### PR TITLE
Make host delete chunk size configurable - RHCLOUD-8272

### PR DESCRIPTION
**Pull request by tahmidefaz**

### CODE REVIEW

File: lib/host_delete.py [hunk 1]
```diff
@@ -9,12 +10,11 @@
 from lib.metrics import delete_host_processing_time
 
 __all__ = ("delete_hosts",)
-CHUNK_SIZE = 1000
 
 
 def delete_hosts(select_query, event_producer, interrupt=lambda: False):
     while select_query.count():
-        for host in select_query.limit(CHUNK_SIZE):
+        for host in select_query.limit(inventory_config().host_delete_chunk_size):
```
Comment (jharting) [inline lib/host_delete.py:17]: This relies on the app context which is not available when reaper runs. As a result this change breaks reaper. We may need to pass the value as a parameter instead.

**Review by jharting — CHANGES_REQUESTED** [review]

File: lib/host_delete.py (hunk 1 above)
Comment (Glutexo) [reply lib/host_delete.py]: It may be worth thinking about some “global” configuration context for values common to all environments.

**Comment by tahmidefaz** [comment]
Finally added that test, any suggestion on improvements?

**Comment by jharting** [comment]
@tahmidefaz if I undo all the changes in this PR except for the test it still passes. That may be an indication that the test is not very good.

**Review by jharting — APPROVED** [review]

**Pull request merged**